File: tools/datarunner/datarunner/review.py
from __future__ import annotations

from typing import Any


def format_table(prices: list[dict[str, Any]]) -> str:
    if not prices:
        return "(no rows parsed)"
    lines = [f"{'name':<24} {'buy':>10} {'sell':>10} {'scu_b':>8} {'scu_s':>8}"]
    lines.append("-" * 64)
    for p in prices:
        lines.append(
            f"{str(p.get('name') or ''):<24} "
            f"{_fmt(p.get('price_buy')):>10} "
            f"{_fmt(p.get('price_sell')):>10} "
            f"{_fmt(p.get('scu_buy')):>8} "
            f"{_fmt(p.get('scu_sell')):>8}"
        )
    return "\n".join(lines)


def _fmt(v: object) -> str:
    if v is None:
        return "-"
    if isinstance(v, float) and v.is_integer():
        return str(int(v))
    return str(v)


def confirm(prompt: str, *, yes: bool) -> bool:
    if yes:
        return True
    try:
        ans = input(f"{prompt} [y/N] ").strip().lower()
    except EOFError:
        return False
    return ans in ("y", "yes")

File: tools/datarunner/src/review.rs
use crate::parse::PriceRow;

pub fn format_table(prices: &[PriceRow]) -> String {
    if prices.is_empty() {
        return "(no rows parsed)".into();
    }
    let mut lines = vec![format!(
        "{:<24} {:>10} {:>10} {:>8} {:>8}",
        "name", "buy", "sell", "scu_b", "scu_s"
    )];
    lines.push("-".repeat(64));
    for p in prices {
        lines.push(format!(
            "{:<24} {:>10} {:>10} {:>8} {:>8}",
            p.name,
            fmt(p.price_buy),
            fmt(p.price_sell),
            fmt_i(p.scu_buy),
            fmt_i(p.scu_sell),
        ));
    }
    lines.join("\n")
}

fn fmt(v: Option<f64>) -> String {
    match v {
        None => "-".into(),
        Some(n) if n.fract() == 0.0 => format!("{}", n as i64),
        Some(n) => format!("{n}"),
    }
}

fn fmt_i(v: Option<i64>) -> String {
    v.map(|n| n.to_string()).unwrap_or_else(|| "-".into())
}

pub fn confirm(prompt: &str, yes: bool) -> bool {
    if yes {
        return true;
    }
    eprint!("{prompt} [y/N] ");
    let mut buf = String::new();
    if std::io::stdin().read_line(&mut buf).is_err() {
        return false;
    }
    matches!(buf.trim().to_ascii_lowercase().as_str(), "y" | "yes")
}
